packages/ckeditor5-font/tests/ui/colorui.js

```diff
@@ -535,4 +535,61 @@ describe( 'ColorUI', () => {
 			expect( dropdown.colorSelectorView.colorPickerView ).to.be.undefined;
 		} );
 	} );
+
+	// Issue: https://github.com/ckeditor/ckeditor5/issues/15580
+	// For simplicity we create editor with two same buttons in toolbar
+	// instead overcomplicating stuff with ballon toolbar.
+	describe( 'toolabar with two same instance of testColor', () => {
+		let editor, element;
+
+		beforeEach( () => {
+			element = document.createElement( 'div' );
+			document.body.appendChild( element );
+
+			return ClassicTestEditor
+				.create( element, {
+					plugins: [ Paragraph, TestColorPlugin, Undo ],
+					testColor: testColorConfig,
+					toolbar: [
+						'testColor',
+						'testColor'
+					]
+				} )
+				.then( newEditor => {
+					editor = newEditor;
+					model = editor.model;
+					testColorPlugin = newEditor.plugins.get( 'TestColorPlugin' );
+				} );
+		} );
+
+		afterEach( () => {
+			element.remove();
+
+			return editor.destroy();
+		} );
+
+		describe( 'testColor Dropdown', () => {
+			let dropdown;
+
+			beforeEach( () => {
+				command = editor.commands.get( 'testColorCommand' );
+				dropdown = editor.ui.componentFactory.create( 'testColor' );
+				dropdown.isOpen = true;
```

Comment:
Unfortunately if you test it like this, then this test will not actually verify the changes you've made in the `colorui.ts`. You can check this by undoing your changes in `colorui.ts` and running this test - it still passes.

To check that your changes fix the problem, you have to either keep two dropdowns in the toolbar and open the first one (instead of creating here a third dropdown) or create here 2 dropdowns and try to open the first one.

Your test does not verify the fix because in these 2 lines you create a dropdown and open the same, so in `colorui.ts` this last reference would be stored and in the console there would be no error. For the error to be displayed you'd have to try to open a dropdown that was created not as the last one.